anvil/src/udev.rs

```diff
@@ -508,6 +518,60 @@ pub fn run_udev() {
     }
 }
 
+impl DrmLeaseHandler for AnvilState<UdevData> {
+    fn drm_lease_state(&mut self, node: DrmNode) -> &mut DrmLeaseState {
+        self.backend_data
+            .backends
+            .get_mut(&node)
+            .unwrap()
+            .leasing_global
+            .as_mut()
+            .unwrap()
+    }
+
+    fn lease_request(
+        &mut self,
+        node: DrmNode,
+        request: DrmLeaseRequest,
+    ) -> Result<DrmLeaseBuilder, LeaseRejected> {
+        let backend = self.backend_data.backends.get(&node).unwrap();
```

Comment:
Could this result in a race when the device gets removed after the client issues a lease request but before the server processing the request?

Reply:
Yeah probably. I guess I need to change the `unwrap` into a `return LeaseRejected` for the error case. Nice catch.